src/main/java/com/redhat/lightblue/camel/LightblueProducer.java

```diff
@@ -22,7 +24,20 @@ public LightblueProducer(LightblueScheduledPollEndpoint endpoint) {
     public void process(Exchange exchange) throws Exception {
         LightblueRequest req = (AbstractLightblueDataRequest) exchange.getIn().getBody();
 
-        exchange.getOut().setBody(endpoint.getLightblueClient().data(req));
+        try {
+            LightblueResponse response = endpoint.getLightblueClient().data(req);
+            if (response.hasError()) {
+                throw new RuntimeException("LightblueProducer:Error returned in response: " + response.getText());
```

Comment:
Why not just set the exchange exception here rather than throwing this exception only to have it caught a few lines down.